subsys/net/lib/sockets/sockets.c

```diff
@@ -1191,6 +1191,88 @@ int zsock_wait_data(struct net_context *ctx, k_timeout_t *timeout)
 	return 0;
 }
 
+
+static int insert_pktinfo(struct msghdr *msg, int level, int type,
+			  void *pktinfo, size_t pktinfo_len)
+{
+	struct cmsghdr *cmsg;
+
+	cmsg = CMSG_FIRSTHDR(msg);
```

Comment:
Shall `cmsg` be checked for NULL ptr? `CMSG_FIRSTHDR` will return NULL if `msg->msg_controllen < sizeof(struct cmsghdr)`